### PR TITLE
Add ControlledBy widgets, fix empty group bug

### PR DESCRIPTION
**Pull request by dariadomagala**
**Description**

Changes proposed in this pull request:

- Add controlledBy widgets + docs
- Fix empty group bug
- ...

To test it, you can use draft configuration for services from [here](https://github.com/kyma-project/busola/blob/d35b2c79f0fc7fa8c0d065c0fe001ba5970bedfb/core/src/assets/customResources/customResources.json)
**Related issue(s)**

Resolves https://github.com/kyma-project/busola/issues/1348 (bugs discovered by this spike 🙂)

### CODE REVIEW

File: core-ui/src/shared/components/ControlledBy/ControlledBy.js [hunk 1]
```diff
@@ -99,43 +99,69 @@ export const GoToDetailsLink = ({
...
 export const ControlledBy = ({ ownerReferences }) => {
-  if (!ownerReferences?.length) return EMPTY_TEXT_PLACEHOLDER;
+  if (
+    !ownerReferences ||
+    (Array.isArray(ownerReferences) && !ownerReferences?.length)
+  )
+    return EMPTY_TEXT_PLACEHOLDER;
+
+  const OwnerRef = ({ owner, className }) => {
+    const resource = owner.kind.endsWith('s')
```
Comment (wchrapka) [inline core-ui/src/shared/components/ControlledBy/ControlledBy.js:111]: it probably works, but don't we use pluralize for that throughout the app?

**Review by wchrapka — APPROVED** [review]

File: docs/extensibility/display-widgets.md [hunk 2]
```diff
@@ -139,3 +139,33 @@ Additionally, you must define the kind of the linked resources by passing the Ku
   "kind": "Secret"
 }
 ```
+
+### ControlledBy
+
+ControlledBy widgets render the kind and the name with a link to the resources that the current resource is dependent of.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:145]:
```suggestion
ControlledBy widgets render the kind and the name with a link to the resources that the current resource is dependent on.
```

File: docs/extensibility/display-widgets.md [hunk 3]
```diff
@@ -139,3 +139,33 @@ Additionally, you must define the kind of the linked resources by passing the Ku
...
+  "path": "metadata.ownerReferences",
+  "widget": "ControlledBy"
+}
+```
+
+<img src="./assets/display-widgets/ControlledBy.png" alt="Example of a table widget" width="20%" style="border: 1px solid #D2D5D9">
+
+### ControlledByKind
+
+ControlledByKind widgets render the kind of the resources that the current resource is dependent of.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:160]:
```suggestion
ControlledByKind widgets render the kind of the resources that the current resource is dependent on.
```

**Review by wchrapka — APPROVED** [review]

**Pull request merged**